http/src/main/java/io/micronaut/http/MediaType.java

```diff
@@ -532,9 +532,9 @@ public CharSequence subSequence(int start, int end) {
      * @return Whether the media type is text based
      */
     public boolean isTextBased() {
-        boolean matches = textTypePatterns.stream().anyMatch((p) -> p.matcher(name).matches());
+        boolean matches = textTypePatterns.stream().anyMatch(p -> p.matcher(name).matches());
         if (!matches) {
-            matches = subtype.equalsIgnoreCase("json") || subtype.equalsIgnoreCase("xml");
+            matches = subtype.equalsIgnoreCase("json") || subtype.equalsIgnoreCase("xml") || subtype.equalsIgnoreCase("x-yaml");
```

Comment:
Is there a reason why only `x-yaml` is checked?

Reply:
Well the media type is `application/x-yaml` and `text/yaml` will be matched by the `textTypePatterns`.

Reply:
I can add a pattern if you want:
```java
textTypePatterns.add(Pattern.compile("^.*\\+yaml$"));
```